pkg/objects/debian.go

```diff
@@ -265,63 +272,90 @@ func (s *DebianSource) ListDistFiles(distRootPath string, cache map[string]FileS
 		}
 
 		//note architecture specific files
-		if len(s.Architectures) == 0 {
+		for _, arch := range architectures {
+			//note 'Contents' indices
 			switch {
-			//note all 'Contents' indices
 			case debReleaseContentsEntryRx.MatchString(entry.Filename):
-				distFiles = append(distFiles, fileName)
+				matchList := debReleaseContentsEntryRx.FindStringSubmatch(entry.Filename)
+				if matchList[4] == arch {
+					distFiles = append(distFiles, fileName)
+				}
 
-			//note all 'dep11' files
+			//note 'dep11' files
 			case debReleaseDep11EntryRx.MatchString(entry.Filename):
-				distFiles = append(distFiles, fileName)
+				matchList := debReleaseDep11EntryRx.FindStringSubmatch(entry.Filename)
+				if matchList[6] != "" {
+					if matchList[6] == arch {
+						//'dep11' components files
+						distFiles = append(distFiles, fileName)
+					}
+				} else {
+					//'dep11' icon files
+					distFiles = append(distFiles, fileName)
+				}
 
-			//note all 'Packages' indices
+			//note 'Packages' indices
 			case debReleasePackagesEntryRx.MatchString(entry.Filename):
-				distFiles = append(distFiles, fileName)
+				matchList := debReleasePackagesEntryRx.FindStringSubmatch(entry.Filename)
+				if matchList[3] == arch {
+					distFiles = append(distFiles, fileName)
 
-				if exists := packageIndices[stripFileExtension(fileName)]; !exists {
-					packageIndices[stripFileExtension(fileName)] = true
+					if exists := packageIndices[stripFileExtension(fileName)]; !exists {
+						packageIndices[stripFileExtension(fileName)] = true
+					}
 				}
 			}
+		}
+	}
 
-		} else {
-			//if config file specifies architectures then only the respective
-			//files are noted
-			for _, arch := range s.Architectures {
-				//note 'Contents' indices
-				switch {
-				case debReleaseContentsEntryRx.MatchString(entry.Filename):
-					matchList := debReleaseContentsEntryRx.FindStringSubmatch(entry.Filename)
-					if matchList[4] == arch {
-						distFiles = append(distFiles, fileName)
-					}
+	//note other miscellanous files that exist under the main component
+	//for some $DIST
+	for _, arch := range architectures {
+		//get a file listing for '$DIST_ROOT/main/installer-$ARCH/'
+		entries, lerr := s.recursivelyListEntries(filepath.Join(distRootPath, "main", "installer-"+arch))
+		if lerr != nil {
+			if !strings.HasSuffix(lerr.Message, "GET returned status 404 Not Found") {
```

Comment:
We should not rely on the words "Not Found" here (and below). Webservers are free to provide arbitrary status texts. A check for `strings.Contains(..., "GET returned status 404")` should be enough.